pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -456,6 +458,75 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, err
 	}
 
+	if cd.Spec.MachineManagementStrategy != nil && cd.Spec.MachineManagementStrategy.Strategy == "Central" {
+		targetNamespace := cd.Status.TargetNamespace
+		if targetNamespace == "" {
+			targetNamespace = apihelpers.GetResourceName(cd.Name+"-targetns", utilrand.String(5))
+		}
+
+		cdLog.Info("Creating the target namespace ", targetNamespace)
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: targetNamespace},
+		}
+		if err := r.Create(context.TODO(), ns); err != nil && !apierrors.IsAlreadyExists(err) {
+			return reconcile.Result{}, fmt.Errorf("failed to create target namespace %q: %w", ns.Name, err)
+		}
+
+		if cd.Status.TargetNamespace == "" {
+			cd.Status.TargetNamespace = targetNamespace
+			if err := r.Status().Update(context.TODO(), cd); err != nil {
+				cdLog.WithError(err).Log(controllerutils.LogLevel(err), "could not set cluster target namespace")
+				return reconcile.Result{Requeue: true}, nil
+			}
+		}
+
+		// Add cluster deployment as additional owner reference to namespace and set clusterName annotation
+		if err := r.addOwnershipToTargetNamespace(cd, cdLog, targetNamespace); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		cdLog.Infof("Creating credentials secret in the target namespace %s", targetNamespace)
+		credentialsSecretName := utils.CredentialsSecretName(cd)
+		credentialsSecret := &corev1.Secret{}
+		err := r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: credentialsSecretName}, credentialsSecret)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to get provider creds %s: %w", credentialsSecretName, err)
+		}
+		credentialsSecret.Namespace = targetNamespace
+		credentialsSecret.ResourceVersion = ""
+		if err := r.Create(context.TODO(), credentialsSecret); err != nil && !apierrors.IsAlreadyExists(err) {
+			return reconcile.Result{}, fmt.Errorf("failed to create provider creds secret: %v", err)
+		}
```

Comment:
It can probably wait but we will likely want to ensure the secret matches the original here so updates trickle out, and that modifying the secret triggers this reconcile.